client/components/app.jsx

```diff
@@ -93,34 +146,126 @@ class App extends React.Component {
     this.setChangeCaption(this.state.defaultChangeCaption);
   }
 
+  handleOptionClick(e) {
+    if (e.target.classList.contains('option')) {
+      this.setState({
+        selectedGraph: e.target.innerHTML
+      });
+    }
+  }
+
   requestData(path, callback) {
-    // Makes a get request to the provided path for a randomly generated stock
+  // Makes a get request to the provided path for a randomly generated stock, and then invokes any provided callback
+
+    // Get a list of all available stocks
     axios.get('/stocks/')
       .then(({data}) => { 
+        // Select a random company from the list
         var randomCompany = data[Math.floor(Math.random() * data.length)];
         this.setCompanyName(randomCompany.name);
         var ticker = randomCompany.ticker;
+        // Get details for the selected company
+        axios.get(`/stocks/${ticker}`)
+          .then(({data}) => {
+            this.setState({
+              analystBuy: (data.analystBuy*100).toFixed(0),
+              platformOwners: data.platformOwners.toLocaleString()
+            });
+          });
+        // Get price history for the selected company
         axios.get(`/stocks/${ticker}/${path}`)
           .then(({data}) => { 
-            this.setLatestPrice(data.last5yPrices[0].price);
-            this.setSelectedPrice(data.last5yPrices[0].price);
-            this.setRefStartPrice(data.last5yPrices[data.last5yPrices.length - 1].price);
-            callback(data.last5yPrices); 
+            this.setTicker(ticker, () => {callback(data)});
           });
       });
   }
 
+  getYesterdayClose(callback) {
+  // Makes a get request specifically for yesterday's close price, and then invokes any provided callback
+    axios.get(`/stocks/${this.state.ticker}/yesterdayClose/`)
+      .then(({data}) => {callback(data)});
+  }
+
   render() {
     var tooltipY = 110;
-    var change = (this.state.selectedPrice - this.state.refStartPrice).toFixed(2);
+    var change = this.state.selectedPrice - this.state.refStartPrice;
     var changePercent = ((this.state.selectedPrice - this.state.refStartPrice) * 100 / this.state.refStartPrice).toFixed(2);
+    var changeAfterHours = this.state.latestAfterHours - this.state.selectedPrice;
+    var changePercentAfterHours = ((this.state.latestAfterHours - this.state.selectedPrice) * 100 / this.state.selectedPrice).toFixed(2);
+    
+    // A second line of price change captions should only show for the 1D graph view
+    var afterHours = <div></div>;
+    if (this.state.changeCaption === 'Today' && this.state.latestAfterHours) {
+      afterHours = 
+        <div>
+          <Change id='changeAfterHours'>
+            {changeAfterHours > 0 ? `+$${Math.abs(changeAfterHours).toFixed(2)}` : `-$${Math.abs(changeAfterHours).toFixed(2)}`} 
+            {' (' + changePercentAfterHours + '%) '}
+          </Change>
+          <ChangeCaption id='changeCaptionAfterHours'>
```

Comment:
airbnb prefers double quotes in jsx https://github.com/airbnb/javascript/tree/master/react#quotes